websites/Y/YNOproject/presence.ts

```diff
@@ -0,0 +1,66 @@
+const presence = new Presence({
+	clientId: "1028080411772977212",
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/2LD3PQV.png",
+		},
+		titleInfo = document.title.split(" - "),
+		location = document.querySelector("#locationText");
+
+	if (titleInfo.length > 1 && titleInfo[1] === "YNOproject" ? true : false) {
+		switch (
+			document.querySelector("#modalContainer .modal:not(.hidden) .modalTitle")
+				?.textContent
+		) {
+			case "Settings":
+				presenceData.details = `Changing settings for ${titleInfo[0]}`;
+				break;
+			case "Rankings":
+				presenceData.details = `Viewing rankings for ${titleInfo[0]}`;
+				break;
+			case "Badges":
+				presenceData.details = `Viewing badges for ${titleInfo[0]}`;
+				break;
+			case "Expeditions":
+				presenceData.details = `Viewing expeditions for ${titleInfo[0]}`;
+				break;
+			case "UI Theme":
+				presenceData.details = `Changing UI theme for ${titleInfo[0]}`;
+				break;
+			case "Log In":
+				presenceData.details = `Logging in to ${titleInfo[0]}`;
+				break;
+			default:
+				presenceData.details = `Dreaming on ${titleInfo[0]}`;
+				break;
```

Comment:
Is there a possibility of `details` here getting particularly long when being displayed due to the combination of static and dynamic data? If so, use `PresenceData.state` too

Reply:
Just confirmed it happens on Answered Prayers.
Will fix.

Reply:
Kept the game title as a details and the location/action as the status.
This should keep everything under the max char limit, except if the ingame location has a really long name.